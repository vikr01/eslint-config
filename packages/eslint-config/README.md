# @vikr01/eslint-config
