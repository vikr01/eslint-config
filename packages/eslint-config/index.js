'use strict';

const config = require('./dist');

module.exports = config.default || config;
