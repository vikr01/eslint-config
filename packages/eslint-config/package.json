{
  "name": "@vikr01/eslint-config",
  "license": "MIT",
  "repository": "https://github.com/vikr01/eslint-config/tree/master/packages/eslint-config",
  "version": "9.25.1",
  "type": "commonjs",
  "exports": {
    ".": "./dist/index.js"
  },
  "scripts": {
    "prebuild": "npm run clean",
    "clean": "rimraf dist"
  },
  "dependencies": {
    "@eslint/config-helpers": "^0.2.2",
    "@eslint/js": "^9.25.1",
    "type-fest": "^4.41.0"
  },
  "peerDependencies": {
    "@eslint/json": "^0.12.0",
    "@eslint/markdown": "^6.4.0",
    "eslint": "^9.25.1",
    "eslint-plugin-jest": "^28.11.0",
    "eslint-plugin-prettier": ">=5.2.6",
    "globals": ">=16.0.0"
  },
  "peerDependenciesMeta": {
    "@eslint/json": {
      "optional": true
    },
    "eslint-plugin-jest": {
      "optional": true
    },
    "eslint-plugin-prettier": {
      "optional": true
    }
  },
  "keywords": [
    "eslint",
    "eslintconfig"
  ],
  "engines": {
    "node": ">=20"
  }
}
