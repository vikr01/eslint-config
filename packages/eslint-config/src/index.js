export default {
  extends: ['eslint-config-airbnb-base'].map(require.resolve),
};
