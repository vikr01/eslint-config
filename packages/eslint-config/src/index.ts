/* eslint-disable @typescript-eslint/no-require-imports */
import type { MergeDeep } from "type-fest";

import { defineConfig } from "eslint/config";
import eslintPluginPrettierRecommended from "eslint-plugin-prettier/recommended";
import createJs from "./configs/createJs";
import cjs from "./configs/cjs";
import ignores from "./configs/ignores";
import type * as createTs from "./configs/createTs";
import type * as jsonConfig from "./configs/json";
// import md from './md';

type TypescriptParams =
  | {
      typescript: true;
      tsConfigPath: Parameters<typeof createTs.default>[0]["tsConfigPath"];
    }
  | {
      typescript: false;
    };

type OtherParams = {
  enableIgnores?: false;

  // TODO: support jsConfigPath
  jsConfigPath?: string;

  json?: boolean;
};

type Params = Readonly<MergeDeep<TypescriptParams, OtherParams>>;

export const createConfig = (
  params: Params,
): ReturnType<typeof defineConfig> => {
  const { enableIgnores = true, json: lintJson = false, typescript } = params;
  const jsConfig = createJs({});

  const configs = [
    jsConfig,
    typescript &&
      (require("./configs/createTs") as typeof createTs).default({
        jsConfig,
        tsConfigPath: params.tsConfigPath,
      }),
    cjs,
    // md,
    lintJson && (require("./configs/json") as typeof jsonConfig).default,
    enableIgnores && ignores,
    eslintPluginPrettierRecommended,
  ].filter(
    Boolean as unknown as <T>(x: T) => x is NonNullable<Exclude<T, false>>,
  );

  return defineConfig(...configs);
};
