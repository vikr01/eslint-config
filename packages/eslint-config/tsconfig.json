../shared-configs/tsconfig.symlink.json