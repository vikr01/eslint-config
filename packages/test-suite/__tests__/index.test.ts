import { describe, it, expect } from "vitest";
import { createConfig } from "@vikr01/eslint-config";
import lint from "./helpers/lint";

const mainConfig = createConfig({
  typescript: true,
  tsConfigPath: undefined,
  json: true,
});

describe("config", () => {
  it("main", async () => {
    const report = await lint(mainConfig);
    expect(report).toMatchSnapshot();
  });
});
