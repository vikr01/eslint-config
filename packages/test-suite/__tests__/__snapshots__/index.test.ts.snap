// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`config > main 1`] = `
[
  {
    "errorCount": 0,
    "fatalErrorCount": 0,
    "filePath": "__tests__/helpers/lint.ts",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": [],
    "suppressedMessages": [],
    "usedDeprecatedRules": [],
    "warningCount": 0,
  },
  {
    "errorCount": 0,
    "fatalErrorCount": 0,
    "filePath": "__tests__/index.test.ts",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": [],
    "output": "import { describe, it, expect } from 'vitest';
import { defaultConfig as mainConfig } from '../../eslint-config/src/index';
import lint from './helpers/lint';

describe('config', () => {
  it('main', async () => {
    const report = await lint(mainConfig);
    expect(report).toMatchSnapshot();
  });
});
",
    "suppressedMessages": [],
    "usedDeprecatedRules": [],
    "warningCount": 0,
  },
  {
    "errorCount": 0,
    "fatalErrorCount": 0,
    "filePath": "lib/bar.ts",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": [],
    "suppressedMessages": [],
    "usedDeprecatedRules": [],
    "warningCount": 0,
  },
  {
    "errorCount": 0,
    "fatalErrorCount": 0,
    "filePath": "lib/foo.ts",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": [],
    "suppressedMessages": [],
    "usedDeprecatedRules": [],
    "warningCount": 0,
  },
  {
    "errorCount": 0,
    "fatalErrorCount": 0,
    "filePath": "src/components/Bar.tsx",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": [],
    "suppressedMessages": [
      {
        "column": 14,
        "endColumn": 16,
        "endLine": 8,
        "line": 8,
        "message": "The \`{}\` ("empty object") type allows any non-nullish value, including literals like \`0\` and \`""\`.
- If that's what you want, disable this lint rule with an inline comment or configure the 'allowObjectTypes' rule option.
- If you want a type meaning "any object", you probably want \`object\` instead.
- If you want a type meaning "any value", you probably want \`unknown\` instead.",
        "messageId": "noEmptyObject",
        "nodeType": "TSTypeLiteral",
        "ruleId": "@typescript-eslint/no-empty-object-type",
        "severity": 2,
        "suggestions": [
          {
            "data": {
              "replacement": "object",
            },
            "desc": "Replace \`{}\` with \`object\`.",
            "fix": {
              "range": [
                233,
                235,
              ],
              "text": "object",
            },
            "messageId": "replaceEmptyObjectType",
          },
          {
            "data": {
              "replacement": "unknown",
            },
            "desc": "Replace \`{}\` with \`unknown\`.",
            "fix": {
              "range": [
                233,
                235,
              ],
              "text": "unknown",
            },
            "messageId": "replaceEmptyObjectType",
          },
        ],
        "suppressions": [
          {
            "justification": "",
            "kind": "directive",
          },
        ],
      },
      {
        "column": 14,
        "endColumn": 16,
        "endLine": 10,
        "line": 10,
        "message": "The \`{}\` ("empty object") type allows any non-nullish value, including literals like \`0\` and \`""\`.
- If that's what you want, disable this lint rule with an inline comment or configure the 'allowObjectTypes' rule option.
- If you want a type meaning "any object", you probably want \`object\` instead.
- If you want a type meaning "any value", you probably want \`unknown\` instead.",
        "messageId": "noEmptyObject",
        "nodeType": "TSTypeLiteral",
        "ruleId": "@typescript-eslint/no-empty-object-type",
        "severity": 2,
        "suggestions": [
          {
            "data": {
              "replacement": "object",
            },
            "desc": "Replace \`{}\` with \`object\`.",
            "fix": {
              "range": [
                251,
                253,
              ],
              "text": "object",
            },
            "messageId": "replaceEmptyObjectType",
          },
          {
            "data": {
              "replacement": "unknown",
            },
            "desc": "Replace \`{}\` with \`unknown\`.",
            "fix": {
              "range": [
                251,
                253,
              ],
              "text": "unknown",
            },
            "messageId": "replaceEmptyObjectType",
          },
        ],
        "suppressions": [
          {
            "justification": "",
            "kind": "directive",
          },
        ],
      },
    ],
    "usedDeprecatedRules": [],
    "warningCount": 0,
  },
  {
    "errorCount": 0,
    "fatalErrorCount": 0,
    "filePath": "src/components/Foo.tsx",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": [],
    "suppressedMessages": [],
    "usedDeprecatedRules": [],
    "warningCount": 0,
  },
]
`;
